standards/application/messaging-api.md

```diff
@@ -0,0 +1,965 @@
+allowing them to be terminated when they are no longer needed.
+
+##### `SubscriptionListener`
+A callback function that is invoked whenever a new `MessageRecord` is received.
+It processes the incoming message and MAY return a `Promise` to support asynchronous handling.
+The `MessageRecord` is defined in the `Message Storage API` section.
+
+##### `subscribe`
+Registers a new subscription to listen for messages processed by a given `Decoder`.
+The subscription is based on the `pubsubTopic` and `contentTopic` provided by the `Decoder`,
+which determine the scope of the subscription and the messages to be received.
+The subscription is uniquely identified by a `SubscriptionID`,
+which MUST be used for managing the subscription (e.g., for later unsubscribing).
+The method MUST throw an error if the `pubsubTopic` provided by the `Decoder` is not supported by the underlying node.
+No specific order is guaranteed for callback invocation;
+subscription callbacks MUST be executed as messages are received from the protocol in use.
+If `confirmStored` is set to `true`, a recurring [STORE](../standards/core/store.md) query MUST be started.
+
+##### `unsubscribe`
+Removes a previously created subscription identified by its unique `SubscriptionID`.
+Once unsubscribed, the associated `SubscriptionListener` MUST NOT receive any further messages.
+If the provided `SubscriptionID` does not correspond to an active subscription,
+the call MUST be ignored.
+If recurring [STORE](../standards/core/store.md) query was initiated by this `SubscriptionID`, it MUST be stopped.
+
+##### `getActiveSubscriptions`
+Returns an array of unique `SubscriptionID` values representing all currently active subscriptions.
+Each identifier corresponds to a subscription that is in effect,
+and no duplicate values MUST be present in the returned array.
+
+##### `Decoder`
+The `Decoder` object encapsulates the logic for processing received messages.
+It MUST ensure that incoming messages are decoded from their received format into a usable structure.
+Additionally, it MUST filter messages based on the `contentTopic` to ensure that only relevant messages are processed.
+The specific types of decoders, their extended functionalities,
+and their instantiation mechanisms are considered out of scope for this document.
+
+#### REST API
+
+Reading messages is performed via the `/messages?subscriptionId="<GUID>"` endpoint,
+as detailed in the `Message Storage API` section.
+
+##### `subscribe`
+Request:
+```http
+POST /subscribe HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+
+{
+  "pubsubTopic": "/waku/2/rs/0/2",
+  "contentTopic": "/example/2/rfc/proto"
+}
+```
+
+Response:
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "subscriptionId": "9978c4f1-4465-4537-9d95-0cf016fb1080"
+}
+```
+
+Error for `404`:
+```http
+HTTP/1.1 404 Not Found
+Content-Type: application/json
+
+{
+  "error": "Cannot create subscription. clusterId 0 is not supported."
```

Comment:
There is no `clusterId` in `/subscribe` - this message would utterly confuse me as a developer:) Probably related to my question(s) about replacing `pubsubTopic` with `shard` (and then also adding `clusterid` to `/subscribe` for completeness)